src/drawbot_skia/drawing.py

```diff
@@ -107,6 +108,44 @@ def text(self, txt, position, align=None):
                 self._canvas.scale(1, -1)
             self._drawItem(self._canvas.drawTextBlob, blob, 0, 0)
 
+    def textBox(self, txt, box, fontPath, align=None, boxFill=None):
+        # TODO: better/dedicated lineHeight, alignment, word breaks, 
+        # hyphenation, boxColor, 
+        #
+        # solve fontPath argument
+        #
+        # better lineWidth calcualtion : the last char on the line
+        # should be calculated by boundingBox, not advance width
+        #
+        # textCircle? textSkew? textTriangle? >> textPath..
+        
+        
+        currentFontSize = self.textSize("x")[1]
+        
+        # if boxFill:
+        #     with self.savedState():
+        #         GraphicsStateMixin.fill(*boxFill)
+        #         GraphicsStateMixin.rect(*box)
+        
+        x,y,w,h = box
+
+        while txt:
+            lineWidth = 0
+            lineText = ""
+            while lineWidth < w:
+                if txt:
+                    char = txt[0]
+                    lineWidth += getAdvanceWidth(char,fontPath, currentFontSize)
```

Comment:
I'm not sure I'm in favor of even introducing a textBox implementation that doesn't actually do text shaping. Naive character by character placement of text is only *guaranteed* to break now for most scripts and fonts, and later for for the handful that happened to "work" when actually shaping starts getting done.

Did you look into using [uharfbuzz](https://github.com/harfbuzz/uharfbuzz) instead?